cypress/e2e/models/migration/migration-waves/migration-wave.ts

```diff
@@ -0,0 +1,189 @@
+        cy.url().then(($url) => {
+            if ($url != MigrationWave.fullUrl) {
+                selectUserPerspective("Migration");
+                clickByText(navMenu, migrationWaves);
+                selectItemsPerPage(100);
+            }
+        });
+    }
+
+    public static openNewForm() {
+        MigrationWave.open();
+        clickByText(button, createNewButton);
+    }
+
+    public create() {
+        MigrationWave.openNewForm();
+        this.fillForm(this);
+
+        cy.get(MigrationWaveView.submitButton, { timeout: 10 * SEC })
+            .should("be.enabled")
+            .click();
+    }
+
+    public edit(updateValues: Partial<MigrationWave>) {
+        MigrationWave.open();
+        this.expandActionsMenu();
+        cy.contains(editAction).click();
+
+        this.fillForm(updateValues);
+
+        cy.get(MigrationWaveView.submitButton, { timeout: 10 * SEC })
+            .should("be.enabled")
+            .click({ force: true });
+    }
+
+    public delete() {
+        MigrationWave.open();
+        this.expandActionsMenu();
+        cy.contains(deleteAction).click();
+        click(confirmButton);
+    }
+
+    public static fillName(name: string) {
+        inputText(MigrationWaveView.nameInput, name);
+    }
+
+    private static fillStakeHolder(stakeHolderName: string) {
+        inputText(MigrationWaveView.stakeHoldersInput, stakeHolderName);
+        cy.get("button").contains(stakeHolderName).click();
+    }
+
+    private static fillStakeHolderGroup(stakeHolderGroupName: string) {
+        inputText(MigrationWaveView.stakeHolderGroupsInput, stakeHolderGroupName);
+        cy.get("button").contains(stakeHolderGroupName).click();
+    }
+
+    /**
+     * This method should NOT be used to do assertions, if an invalid date is passed, it'll throw an exception
+     * It selects the date using the picker because it can't be manually entered right now due to bug MTA-706
+     * @param date
+     */
+    public fillStartDate(date: Date) {
+        const nowTime = new Date().setHours(0, 0, 0, 0);
+        date.setHours(0, 0, 0, 0);
+        if (nowTime >= date.getTime()) {
+            expect(
+                true,
+                "Start Date should be greater than actual Date. If you want to assert the validation, enter the date manually"
+            ).to.eq(false);
+        }
+
+        const currentStartDate = this.startDate ? this.startDate : new Date();
+        const currentMonth = currentStartDate.toLocaleString("en-us", { month: "long" });
+        cy.get(MigrationWaveView.startDateInput).next("button").click();
+        MigrationWave.selectDateFromDatePicker(date, currentMonth);
+    }
+
+    /**
+     * This method should NOT be used to do assertions, if an invalid date is passed, it'll throw an exception
+     * It selects the date using the picker because it can't be manually entered right now due to bug MTA-706
+     * @param date
+     */
+    public fillEndDate(date: Date) {
```

Comment:
Can we use the same method to fill both the dates ?
fillDates()

Reply:
There are two methods because the validations and error messages are different, and putting both logics in the same method will create a bit of spaghetti code.

The common logic is in a different method called by both functions.
![image](https://github.com/konveyor/tackle-ui-tests/assets/117646518/76b539f4-8ff1-4470-95d7-17872d619cf2)